pkg/engine/deployment.go

```diff
@@ -133,27 +133,41 @@ type deploymentOptions struct {
 	// creates resources to compare against the current checkpoint state (e.g., by evaluating a program, etc).
 	SourceFunc deploymentSourceFunc
 
-	DOT        bool         // true if we should print the DOT file for this deployment.
-	Events     eventEmitter // the channel to write events from the engine to.
-	Diag       diag.Sink    // the sink to use for diag'ing.
-	StatusDiag diag.Sink    // the sink to use for diag'ing status messages.
-
-	isImport bool            // True if this is an import.
-	imports  []deploy.Import // Resources to import, if this is an import.
-
-	// true if we're executing a refresh.
+	// true if we should print the DOT file for this deployment.
+	DOT bool
+	// the channel to write events from the engine to.
+	Events eventEmitter
+	// the sink to use for diag'ing.
+	Diag diag.Sink
+	// the sink to use for diag'ing status messages.
+	StatusDiag diag.Sink
+
+	// True if this is an import operation.
+	isImport bool
+	// Resources to import, if this is an import.
+	imports []deploy.Import
+
+	// true if this deployment is (only) a refresh operation. This should not be
+	// confused with UpdateOptions.Refresh, which will be true whenever a refresh
+	// is happening as part of an operation (e.g. `up --refresh`).
 	isRefresh bool
+
+	// true if this deployment is a preview/dry-run operation.
+	Preview bool
```

Comment:
We use `DryRun` and `Preview` interchangeably. I've tried to make a clean delineation and feel like long term `Preview` is the destination (?), but could be wrong

Reply:
DryRun might make more sense to distinguish between the preview action (run the program but don't actually do anything) vs other "preview" like modes like a refresh preview, or a destroy preview.